rules/platform.bzl

```diff
@@ -23,22 +18,18 @@ _platform_transition = transition(
 )
 
 def _platform_binary_impl(ctx):
-    source_info = ctx.attr.src[DefaultInfo]
-
-    executable = None
-    if source_info.files_to_run and source_info.files_to_run.executable:
-        command = _vars_script(ctx.attr.env, ctx.attr.run_under, source_info.files_to_run.executable.short_path)
-        executable = ctx.actions.declare_file("{}_{}".format(ctx.file.src.basename, ctx.attr.platform))
-        ctx.actions.write(
-            output = executable,
-            content = command,
-            is_executable = True,
-        )
-
+    platform_sanitized = ctx.attr.platform.replace("/", "_").replace(":", "_")
+    dst = ctx.actions.declare_file("{}-{}".format(paths.basename(ctx.file.src.path), platform_sanitized))
+    src = ctx.file.src
+    ctx.actions.run_shell(
```

Comment:
This action is not using any shell tricks such as pipes, redirects, we should use `actions.run` instead, because

> run invokes an executable directly, which is more efficient and less prone to quoting problems.

https://jayconrod.com/posts/109/writing-bazel-rules--moving-logic-to-execution